vertx-web-client/src/main/java/examples/CachingWebClientExamples.java

```diff
@@ -0,0 +1,63 @@
+package examples;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.client.CachingWebClient;
+import io.vertx.ext.web.client.CachingWebClientOptions;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientSession;
+import io.vertx.ext.web.client.spi.CacheStore;
+import io.vertx.ext.web.client.impl.cache.NoOpCacheStore;
+import java.util.HashSet;
+import java.util.Set;
+
+public class CachingWebClientExamples {
+
+  public void create(Vertx vertx) {
+    WebClient client = WebClient.create(vertx);
+    WebClient cachingWebClient = CachingWebClient.create(client);
+  }
+
+  public void createWithOptions(Vertx vertx) {
+    Set<HttpMethod> cachedMethods = new HashSet<>(2);
+    cachedMethods.add(HttpMethod.GET);
+    cachedMethods.add(HttpMethod.HEAD);
+
+    Set<Integer> cachedStatusCodes = new HashSet<>(1);
+    cachedStatusCodes.add(200);
```

Comment:
use instead `addCachedStatusCode` on options directly

Reply:
I actually wanted to demonstrate reducing the status codes, so I changed this to use `removeCachedStatusCode` directly instead of creating a new set.

Reply:
makes sense